File: 13_Roman_to_Integer/roman_to_integer.c
#include <stdio.h>
#include <string.h>

int romanToInt(char*);

int main(){
    printf("%d\n", romanToInt("III"));
    printf("%d\n", romanToInt("IV"));
    printf("%d\n", romanToInt("IX"));
    printf("%d\n", romanToInt("LVIII"));
    printf("%d\n", romanToInt("MCMXCIV"));

}

int romanToInt(char * s){
    int hashmap[26] = {0};
    hashmap['I' - 'A'] = 1;
    hashmap['V' - 'A'] = 5;
    hashmap['X' - 'A'] = 10;
    hashmap['L' - 'A'] = 50;
    hashmap['C' - 'A'] = 100;
    hashmap['D' - 'A'] = 500;
    hashmap['M' - 'A'] = 1000;
    
    int previous = 0, sum = 0;
    for(int i=0; i<strlen(s); i++){
        if (previous < hashmap[ s[i]- 'A']){
            sum -= previous*2;
        }
        sum += (previous = hashmap[s[i] - 'A']);
    }
    return sum;
}

File: 13_Roman_to_Integer/roman_to_integer.py
#!/usr/bin/env python3

class Solution:
    def romanToInt(self, s: str) -> int:
        data_dict = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000,
        }
        previous = 0
        res_sum = 0
        for val in s:
            if previous < data_dict[val]:
                res_sum -= previous * 2
            res_sum += data_dict[val]
            previous = data_dict[val]
        return res_sum

if __name__ == '__main__':
    a = Solution()
    print(a.romanToInt('III'))
    print(a.romanToInt('IV'))
    print(a.romanToInt('IX'))
    print(a.romanToInt('LVIII'))
    print(a.romanToInt('MCMXCIV'))
